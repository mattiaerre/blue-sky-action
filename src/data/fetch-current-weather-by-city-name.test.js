import { config } from 'dotenv';
import fetch from 'node-fetch';

config();

const CITY_NAME = 'Las Vegas';

describe.skip('fetch current weather by city name', () => {
  test(CITY_NAME, async () => {
    const url = `${
      process.env.OPEN_WEATHER_MAP_BASE_URL
    }/data/2.5/weather?q=${CITY_NAME}&appid=${
      process.env.OPEN_WEATHER_MAP_API_KEY
    }`;
    const weather = await fetch(url)
      .then(response => response.json())
      .then(data => data);

    expect(weather).toMatchSnapshot();
  });
});
