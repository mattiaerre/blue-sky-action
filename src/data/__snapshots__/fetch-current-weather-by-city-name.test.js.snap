// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch current weather by city name Las Vegas 1`] = `
Object {
  "base": "stations",
  "clouds": Object {
    "all": 20,
  },
  "cod": 200,
  "coord": Object {
    "lat": 36.17,
    "lon": -115.14,
  },
  "dt": 1509850560,
  "id": 5506956,
  "main": Object {
    "humidity": 32,
    "pressure": 1011,
    "temp": 292.8,
    "temp_max": 294.15,
    "temp_min": 291.15,
  },
  "name": "Las Vegas",
  "sys": Object {
    "country": "US",
    "id": 2061,
    "message": 0.1643,
    "sunrise": 1509890866,
    "sunset": 1509928804,
    "type": 1,
  },
  "visibility": 16093,
  "weather": Array [
    Object {
      "description": "few clouds",
      "icon": "02n",
      "id": 801,
      "main": "Clouds",
    },
  ],
  "wind": Object {
    "deg": 200,
    "gust": 7.7,
    "speed": 3.6,
  },
}
`;
